### DISCUSSION TITLE
Documentation for imgui

### DISCUSSION THREAD

**Discussion by Rafhub69** [ocornut/imgui#5025] [New Users Build/Link/Run issues ONLY!]
I want to start using it in my program, but I can't find any proper documentation. Is there any for this project?

**Comment by mnesarco** [comment]
There is a wiki, but I like the interactive manual also: https://pthom.github.io/imgui_manual_online/manual/imgui_manual.html

Wiki: https://github.com/ocornut/imgui/wiki

**Comment by PathogenDavid** [comment]
In addition to the resources @mnesarco shared, there's a significant amount of general documentation at the top of [`imgui.cpp`](https://github.com/ocornut/imgui/blob/master/imgui.cpp).

**Comment by Rafhub69** [comment]
Thank you for your help. I want to create something that looks like this:
![WyglądMaszynaW](https://user-images.githubusercontent.com/56224419/154228359-a8ce7a64-de62-44cb-8d5e-3f3420108e9d.png)
 But i don't know how to do it. I managed to create a main menu bar using this code: 
  `c++

if (ImGui::BeginMainMenuBar())

        {
            if (ImGui::BeginMenu("Plik"))
            {
                if (ImGui::BeginMenu("Nowy"))
                {
                    if (ImGui::MenuItem("Program"))
                    {
                        newProgram();
                    }

                    if (ImGui::MenuItem("Rozkaz"))
                    {
                        newOrder();
                    }

                    ImGui::EndMenu();
                }

                if (ImGui::MenuItem("Otwórz...  F3"))
                {
                    //Do something
                }

                if (ImGui::MenuItem("Zapisz      F2"))
                {
                    //Do something
                }

                if (ImGui::MenuItem("Zapisz jako..."))
                {
                    //Do something
                }

                if (ImGui::MenuItem("Wyjście"))
                {
                    break;
                }

                ImGui::EndMenu();
            }

            if (ImGui::BeginMenu("Widok"))
            {
                if (ImGui::MenuItem("Lista okien                        F5"))
                {
                    //Do something
                }

                if (ImGui::MenuItem("Okno edycyjne"))
                {
                    //Do something
                }

                if (ImGui::MenuItem("Lista rozkazów                 F10"))
                {
                    //Do something
                }

                if (ImGui::MenuItem("Obserwowane zmienne"))
                {
                    //Do something
                }

                if (ImGui::MenuItem("Jednostka centralna         F11"))
                {
                    //Do something
                }

                if (ImGui::MenuItem("Konsola we/wy                 F12"))
                {
                    //Do something
                }
                ImGui::EndMenu();
            }

            if (ImGui::BeginMenu("Projekt"))
            {
                if (ImGui::MenuItem("Zapisz listę rozkazów"))
                {
                    //Do something
                }

                if (ImGui::MenuItem("Ładuj listę rozkazów"))
                {
                    //Do something
                }

                if (ImGui::MenuItem("Opcje.."))
                {
                    //Do something
                }

                ImGui::EndMenu();
            }

            if (ImGui::BeginMenu("Wykonaj"))
            {
                if (ImGui::MenuItem("Program      F9"))
                {
                    //Do something
                }

                if (ImGui::MenuItem("Rozkaz        F8"))
                {
                    //Do something
                }

                if (ImGui::MenuItem("Takt             F7"))
                {
                    //Do something
                }

                if (ImGui::MenuItem("Do kursora  F4"))
                {
                    //Do something
                }

                if (ImGui::MenuItem("Reset"))
                {
                    //Do something
                }

                if (ImGui::MenuItem("Przerwij"))
                {
                    //Do something
                }

                ImGui::EndMenu();
            }

            if (ImGui::BeginMenu("Pomoc"))
            {
                if (ImGui::MenuItem("Zawartość"))
                {
                    
                }

                if (ImGui::MenuItem("Szukaj tematu"))
                {
                    //Do something
                }

                if (ImGui::BeginMenu("Zmień wygląd"))
                {
                    if (ImGui::MenuItem("Ciemny schemat"))
                    {
                        backgroundColor = black;
                        ImGui::StyleColorsDark();
                    }

                    if (ImGui::MenuItem("Jasny schemat"))
                    {
                        backgroundColor = white;
                        ImGui::StyleColorsLight();
                    }

                    ImGui::EndMenu();
                }
                ImGui::EndMenu();
            }

            ImGui::Text("%.1f FPS", ImGui::GetIO().Framerate);

            ImGui::EndMainMenuBar();
        }`
But I don't know how to do anything else. Any help with this?

**Reply by thedmd** [reply]
Take closer look at `imgui_demo.cpp`. Find widgets you need, play with them to see how they're used.
After that you can manipulate their placement by changing cursor position.
Custom drawing you can do by manually calling current window `ImDrawList` functions.

**Reply by Rafhub69** [reply]
In all of code that I can't find the option to change position of anything. How can I manually change position of anything?

**Reply by thedmd** [reply]
ImGui draw widget at current screen cursor position. You can manipulate it with `ImGui::GetCursorScreenPos` and `ImGui::SetCursorScreenPos`. Checkout `imgui.h`, every function there has also a few words attached to them.
